### PR TITLE
support for isBusy function

### PR DESCRIPTION
**Pull request by rf**
r @Raynos @jcorbin 

`isBusy` functions take no arguments and return an object with properties `busy` (boolean) and `reason` (string)

### CODE REVIEW

File: node/service-name-handler.js [hunk 1]
```diff
@@ -21,19 +21,37 @@
...
+    }
 }
 
 TChannelServiceNameHandler.prototype.type = 'tchannel.service-name-handler';
 
 TChannelServiceNameHandler.prototype.handleRequest = function handleRequest(req, buildRes) {
     var self = this;
+
+    if (self.isBusy) {
+        var busyInfo = self.isBusy();
```
Comment (Raynos) [inline node/service-name-handler.js]: support returning `null` for performance; creating little objects for every req is going to hurt.

File: node/test/busy.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
```
Comment (jcorbin) [inline node/test/busy.js]: actually include this file from node/test/index.js

File: node/service-name-handler.js (hunk 1 above)
Comment (jcorbin) [inline node/service-name-handler.js]: :+1: , in fact why isn't it just a reason string? YAGNI to the whole info object imo

**Comment by Raynos** [comment]
lgtm

**Pull request merged**